master/buildbot/worker/ec2.py

```diff
@@ -71,7 +73,7 @@ def __init__(self, name, password, instance_type, ami=None,
                 "Use of default value of 'keypair_name' of EC2LatentWorker "
                 "constructor is deprecated. Please explicitly specify value")
             keypair_name = 'latent_buildbot_slave'
-        if security_name is None:
+        if security_name is None and not subnet_id:
             reportDeprecatedWorkerNameUsage(
```

Comment:
wow, I wonder how this is related to transition to worker... I think it should be just `DeprecationWarning` usage.  @rutsky?